doc/dev_guide.rst

```diff
@@ -164,3 +164,32 @@ to use it with examples and links to the relevant code.
 When you've got a branch that's ready to be incorporated in to the main code of
 HyperSpy -- make a pull request on GitHub and wait for it to be reviewed and
 discussed.
+
+6. Contributing cython code
+---------------------------
+
+Python is not the fastest language, it is slow in loops. Sometimes using vectorisation
+of calculations with numpy is not possible. It is mostly actual for some proprietary
+binary electron microscopy formats. In such and similar cases where the speed can be
```

Comment:
"It is mostly actual for some proprietary binary electron microscopy formats" doesn't make sense, and there may be cases in future that have nothing to do with binary file formats where Cython may still be helpful - I suggest taking this out.